laws/src/main/scala/cats/laws/MonadDeferLaws.scala

```diff
@@ -0,0 +1,62 @@
+package cats
+package laws
+
+import cats.syntax.all._
+import cats.laws.MonadDeferLaws.StatefulBox
+
+trait MonadDeferLaws[F[_]] extends ApplicativeEvalLaws[F] with MonadLaws[F] {
+  implicit override def F: MonadDefer[F]
+
+  def evalSequenceConsistentWithPureMap[A, B](a: A, b: B, f: (A, B) => A): IsEq[F[A]] = {
+    def tr(s: StatefulBox[A]) = s.transform(a => f(a, b))
+    val state1 = new StatefulBox(a)
+    val state2 = new StatefulBox(a)
+
+    val lh = F.delay(tr(state1))
+    val rh = F.pure(state2).map(tr)
+
+    lh.flatMap(_ => lh) <-> rh.flatMap(_ => rh)
```

Comment:
The triggered code is side-effectul and thought it would be a good idea to test that `delay(a)` is equivalent with `pure.map(_ => a)` in the presence of side-effects. This doesn't necessarily have to be so. Consider ...

```scala
implicit object EvalInstances extends MonadDefer[Eval] {
  def eval[A](a: Eval[A]) = a.memoize
}
```

And now this law is violated.